src/data.py

```diff
@@ -1,101 +1,297 @@
 
-    Parameters:
-    subsets_path (str): The file path to the CSV containing the subsets data.
-    labels_path (str): The file path to the CSV containing the labels data.
+        Parameters:
+        ----------
+        index : int
+            Index of the batch.
 
-    Returns:
-    pd.DataFrame: A merged DataFrame containing data from both input CSVs,
-                  joined on the "image_id" column.
-    """
-    subsets = pd.read_csv(subsets_path)
-    labels = pd.read_csv(labels_path)
-    return pd.merge(subsets, labels, how="inner", on="image_id")
+        Returns:
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            A batch of images and their corresponding labels.
+        """
+        indexes = self.indexes[
+            index * self.batch_size : (index + 1) * self.batch_size
+        ]
 
+        list_IDs_temp = [self.list_IDs[k] for k in indexes]
 
-def create_data_subsets(df: pd.DataFrame) -> None:
-    """
-    Splits the provided DataFrame into training, validation, and test datasets, and saves them as CSV files.
-
-    Args:
-        df (pd.DataFrame): A DataFrame containing the dataset, which includes image IDs, partition labels,
-        and other relevant data.
-
-    Returns:
-        None: This function does not return any value but saves three CSV files: `train.csv`, `validation.csv`,
-        and `test.csv`.
-
-    This function processes the input DataFrame by first filtering the data into training and test sets
-    based on the `partition` column.
-    The training set (where `partition` is 0) is further split into training and validation subsets,
-    with 9% of the data allocated
-    to the validation set. The resulting subsets are then saved as separate CSV files:
-    - `train.csv` for the training data
-    - `validation.csv` for the validation data
-    - `test.csv` for the test data
-
-    The function also prints the number of samples in each of these subsets.
-    """
+        X, y = self.__data_generation(list_IDs_temp)
+
+        return X, y
+
+    def on_epoch_end(self):
+        """
+        Updates indexes after each epoch.
+
+        This method is called at the end of each epoch and is responsible for
+        shuffling the data if the shuffle attribute is set to True.
+        """
+        self.indexes = np.arange(len(self.list_IDs))
+        if self.shuffle:
+            np.random.shuffle(self.indexes)
+
+    def __data_generation(
+        self, list_IDs_temp: List[str]
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Generates data containing batch_size samples.
+
+        Parameters:
+        ----------
+        list_IDs_temp : List[str]
+            List of image IDs to generate data for.
+
+        Returns:
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            A batch of images and their corresponding labels.
+        """
+        X = np.empty(
+            (self.batch_size, self.dim[1], self.dim[0], self.n_channels)
```

Comment:
now I used * but changing dims order is still needed to avoid error. I did it like this:
image = cv2.resize(image, self.dim[::-1])
# avoiding ValueError by adjusting self.dim order